subsys/llext/llext_load.c

```diff
@@ -36,8 +36,21 @@ LOG_MODULE_DECLARE(llext, CONFIG_LLEXT_LOG_LEVEL);
  *   calls, taking into account if the load process was successful or not.
  */
 
+#define LLEXT_SECTION_MIN_ADDR 0x10000
```

Comment:
This hardcoded value which is neither arch- nor even project- independent. If comparing addresses is required, then this must at least be customizable by the caller.

Reply:
@pillo79 agree, yes, I was thinking about it too. Would it be ok for you to pass this as a value in `struct llext_load_param`? Otherwise we could add a platform callback to that structure to perform any testing and return `true` or `false` for whether the symbol is used where it's located or where it's linked for, but I think that would make it uglier, less transparent.

Reply:
I personally see it as the opposite - it would neatly decouple the generic implementation from project-specific reasoning.
For example, I can already see places where I would add callbacks for: 
* "do we allocate a copy of a section?" <- would replace and extend `STORAGE_WRITABLE`, e.g. allow `rodata` from Flash
* "how do we relocate a section?" <- would extend (or even replace!) `pre_located` and simplify the current PR
* "what is the effective base address of this section?" <- would possibly replace link scripts and also be a first step for in-Flash XIP.